liblocation/src/main/java/com/mapbox/android/core/location/LocationIntentHandler.java

```diff
@@ -0,0 +1,34 @@
+package com.mapbox.android.core.location;
+
+import android.content.Intent;
+import android.location.Location;
+
+import com.google.android.gms.location.LocationResult;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+class LocationIntentHandler {
+  private CopyOnWriteArrayList<LocationEngineListener> locationListeners;
+
+  LocationIntentHandler(CopyOnWriteArrayList<LocationEngineListener> locationListeners) {
+    this.locationListeners = locationListeners;
+  }
+
+  void handle(Intent intent, String actionTag) {
+    if (intent != null) {
+      final String action = intent.getAction();
+      if (actionTag.equals(action)) {
```

Comment:
I believe we don't need to compare action tags here, this should be handled by the `IntentFilter`.